src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs

```diff
@@ -0,0 +1,194 @@
+            IThreadingContext threadingContext,
+            EventListenerTracker<ITodoListProvider> eventListenerTracker)
+            : base(threadingContext)
+        {
+            _workspace = workspace;
+            _eventListenerTracker = eventListenerTracker;
+        }
+
+        void IVisualStudioTodoCommentsService.Start(CancellationToken cancellationToken)
+            => _ = StartAsync(cancellationToken);
+
+        private async Task StartAsync(CancellationToken cancellationToken)
+        {
+            // Have to catch all exceptions coming through here as this is called from a
+            // fire-and-forget method and we want to make sure nothing leaks out.
+            try
+            {
+                await StartWorkerAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException)
+            {
+                // Cancellation is normal (during VS closing).  Just ignore.
+            }
+            catch (Exception e) when (FatalError.ReportWithoutCrash(e))
+            {
+                // Otherwise report a watson for any other exception.  Don't bring down VS.  This is
+                // a BG service we don't want impacting the user experience.
+            }
+        }
+
+        private async Task StartWorkerAsync(CancellationToken cancellationToken)
+        {
+            _workQueue = new AsyncBatchingWorkQueue<DocumentAndComments>(
+                TimeSpan.FromSeconds(1),
+                ProcessTodoCommentInfosAsync,
+                cancellationToken);
+
+            var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
+            if (client == null)
+                return;
+
+            // Pass ourselves in as the callback target for the OOP service.  As it discovers
+            // todo comments it will call back into us to notify VS about it.
+            _keepAliveSession = await client.TryCreateKeepAliveSessionAsync(
+                WellKnownServiceHubServices.RemoteTodoCommentsService,
+                callbackTarget: this, cancellationToken).ConfigureAwait(false);
+            if (_keepAliveSession == null)
+                return;
+
+            // Now that we've started, let the VS todo list know to start listening to us
+            _eventListenerTracker.EnsureEventListener(_workspace, this);
+
+            // Now kick off scanning in the OOP process.
+            var success = await _keepAliveSession.TryInvokeAsync(
+                nameof(IRemoteTodoCommentsService.ComputeTodoCommentsAsync),
+                solution: null,
+                arguments: Array.Empty<object>(),
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Callback from the OOP service back into us.
+        /// </summary>
+        Task ITodoCommentsListener.ReportTodoCommentDataAsync(DocumentId documentId, List<TodoCommentData> infos, CancellationToken cancellationToken)
```

Comment:
>ITodoCommentsListener [](start = 13, length = 21)

Have you checked explicit iface implementations work with JSON RPC invocation?

Reply:
nope!

Reply:
they do not :)